crowdfunding/users/views.py

```diff
@@ -0,0 +1,73 @@
+# from django.shortcuts import render
+# used for templates / HTML
+
+# # Create your views here.
+
+from django.http import Http404
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import status, generics, permissions
+from rest_framework.permissions import IsAuthenticated #added 23/1
+from rest_framework import validators #uniquevalidator error handling
+
+from .models import CustomUser
+from .serializers import CustomUserSerializer, ChangePasswordSerializer, CustomUserDetail
+from projects.permissions import IsOwnProfile
+
+class CustomUserList(APIView):
+
+    def get(self, request):
+        users = CustomUser.objects.all()
+        serializer = CustomUserSerializer(users, many=True)
+        return Response(serializer.data)
+        #same as project
+
+    def post(self, request):
+        serializer = CustomUserSerializer(data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        else:
+            if 'email' in serializer.errors:
+                return Response({"error":"This email is associated with another user. Please login or choose an alternative email."}, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
shouldn't this be raised by the serializer already? 
it seems strange you'd have to catch it and then rephrase it